pkg/resource/deploy/deploytest/sink.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deploytest
+
+import (
+	"github.com/pulumi/pulumi/sdk/v3/go/common/diag"
+)
+
+type NoopSink struct {
```

Comment:
More of a MockSink than a NoopSink given the LogfF function to override

Reply:
I would expect a `MockSink` to panic if methods are called and no implementation was provided.

I added this so that if a struct required a Sink interface as a field, you could provide `NoopSink{}` without drawing attention away by writing about/disregarding logging expectations and it would not panic when one of its methods is called because the test was aiming to target a certain behavior. Such as in `pluginhost_test.go`

```go
hostEngine := &hostEngine{
	sink:       &NoopSink{},
	statusSink: &NoopSink{},
}
```

Reply:
hmm yeh it's a bit of sadness of this interface that it has all the other methods on the interface itself even though they only have one sensible implementation. Something to fix down the line, this can be a bit ugly till then.